sklearn/metrics/_regression.py

```diff
@@ -1026,8 +1028,56 @@ def mean_tweedie_deviance(y_true, y_pred, *, sample_weight=None, power=0):
         sample_weight = column_or_1d(sample_weight)
         sample_weight = sample_weight[:, np.newaxis]
 
-    dist = TweedieDistribution(power=power)
-    dev = dist.unit_deviance(y_true, y_pred, check_input=True)
+    p = check_scalar(
+        power,
+        name="power",
+        target_type=numbers.Real,
+    )
+
+    message = f"Mean Tweedie deviance error with power={p} can only be used on "
```

Comment:
From here on, it is really copy&paste from the now deleted `glm_distribution.py`.